src/tqec/generation/circuit.py

```diff
@@ -6,11 +6,11 @@
 from tqec.plaquette.plaquette import Plaquette
 from tqec.plaquette.schedule import ScheduledCircuit, merge_scheduled_circuits
 from tqec.position import Shape2D
-from tqec.templates.orchestrator import TemplateOrchestrator
+from tqec.templates.base import Template
 
 
 def generate_circuit(
-    template: TemplateOrchestrator,
+    template: Template,
     plaquettes: list[Plaquette],
```

Comment:
We can make the type of the plaquettes as `list[Plaqutte] | dict[int, Plaqutte]` to make it more flexible.

Reply:
Yes, I also thought about that when starting to use the `TemplateStack` in #100. I agree, and it should not be too hard to add.

Reply:
Note to myself: the `TemplateStack` requires to provide empty plaquettes anyways as it does not simplify out the plaquettes. This idea is still good, but is more about "future proofing" and less about immediate need, except if I miss something.

Reply:
Should be done!